src/main/scala/sam/sceval/EvaluationPimps.scala

```diff
@@ -243,6 +257,9 @@ object EvaluationPimps extends Logging {
         Iterator(agg)
       }
       .collect()
```

Comment:
@gm-spacagna  No, as explained before I do think this clone is necessary.  If you follow the code through all mutable operations are partionwise and then clones happen prior to shuffles. By cloning we can be sure we are not going to cause any multi-threading issues.  Also, as explained last time you raised this, this isn't my code, and I've deprecated it.  Adding unit tests to check this isn't worth the hassle.

If you PR the original MLLib code to remove it, and they accept it, THEN I will remove it from here :) and good luck with that :)